Add truncate description to 50 characters

LatifaBik opened the pull request:
* This pull request introduces a truncation function that shortens challenge descriptions to a maximum of 50 characters, ensuring clean and consistent card layouts.
*  Because the challenge descriptions from the API are already short, an automated test will be added to verify that the truncation function behaves as expected.

latifabikndarne committed:
Add truncate description to 50 characters

latifabikndarne committed:
an automated test is added to verify that the truncateDescription function

latifabikndarne committed:
Add new file för truncateDescription.js instead to use main.js

leolinder-hash reviewed (changes requested):
Hi Latifa! First of all, great work with the truncation function. The code looks great and it is also nice that the descriptions are prevented from splitting mid-sentence. 

However, I noticed that the function doesn't run correctly because of issues with the export. I fixed this locally, so before merge i suggest these changes:

1) Remove module.exports = { truncateDescription }; from truncateDescription.js
2) Add "export" to the function like this: export function truncateDescription(description, maxLength = 50) {
3) Import the function in main.js with: import { truncateDescription } from "./truncateDescription.js";     (put it at the top of main.js
4) Implement it in main.js line 93 with:
      <p class="challenge__description">${truncateDescription(description)}</p>

After this it should be ready to merge, really well done! :)

latifabikndarne committed:
Fixed issues with the export of function truncateDescription

LatifaBik commented:
Hi Leo! Thank you for your detailed feedback 🙂
I have now implemented the changes you suggested:

1.  Removed module.exports = { truncateDescription }; from truncateDescription.js
2.  Added export to the function declaration
3.  Imported the function in main.js using: import { truncateDescription } from "./truncateDescription.js"; 
4.  Implemented the function in main.js: ${truncateDescription(description)}

And also Updated package.json from "type": "commonjs" to "type": "module"

Everything should now be updated according to your review. 
Thanks again!

latifabikndarne committed:
Moved the truncateDescription function call below the description element.

latifabikndarne committed:
Removed truncateDescription outside the p element

leolinder-hash reviewed (approved):
Hey again Latifa! I've now looked at your changes and everything looks perfect. I'll approve this for testing. Great work :)

The only thing now is a small conflict with main in main.css.map. It should be enough to fix this if you:
- include main.css.map in your gitignore-file
- Make a new push to this branch

Again, great job!

latifabikndarne committed:
main.css.map is included in gitignore-file

LatifaBik committed:
Merge branch 'main' into Max_50_characters_for_card_description

Kevinekmen commented:
I tested this and it works as expected. Well done Latifa!
